keras_cv/layers/preprocessing3d/swap_background.py

```diff
@@ -0,0 +1,164 @@
+        objects_points_in_point_clouds = tf.reduce_any(
+            is_within_box3d(point_clouds[:, :, :3], bounding_boxes[:, :, :7]),
+            axis=-1,
+            keepdims=True,
+        )
+        point_clouds = tf.where(~objects_points_in_point_clouds, point_clouds, 0.0)
+
+        # Extract objects from additional_point_clouds.
+        objects_points_in_additional_point_clouds = tf.reduce_any(
+            is_within_box3d(
+                additional_point_clouds[:, :, :3], additional_bounding_boxes[:, :, :7]
+            ),
+            axis=-1,
+            keepdims=True,
+        )
+        additional_point_clouds = tf.where(
+            objects_points_in_additional_point_clouds, additional_point_clouds, 0.0
+        )
+
+        # Remove backgorund points in point_clouds overlaps with additional_bounding_boxes.
+        points_overlaps_additional_bounding_boxes = tf.reduce_any(
+            is_within_box3d(
+                point_clouds[:, :, :3], additional_bounding_boxes[:, :, :7]
+            ),
+            axis=-1,
+            keepdims=True,
+        )
+        point_clouds = tf.where(
+            ~points_overlaps_additional_bounding_boxes, point_clouds, 0.0
+        )
+        return {
+            POINT_CLOUDS: point_clouds,
+            ADDITIONAL_POINT_CLOUDS: additional_point_clouds,
+            ADDITIONAL_BOUNDING_BOXES: additional_bounding_boxes,
+        }
+
+    def augment_point_clouds_bounding_boxes(
+        self, point_clouds, bounding_boxes, transformation, **kwargs
+    ):
+        original_bounding_boxes_shape = bounding_boxes.get_shape().as_list()
+        original_point_clouds_shape = point_clouds.get_shape().as_list()
+        point_clouds = transformation[POINT_CLOUDS]
+        additional_point_clouds = transformation[ADDITIONAL_POINT_CLOUDS]
+        num_frames = original_point_clouds_shape[0]
+        point_clouds_list = []
+        for frame_index in range(num_frames):
+            background_point_clouds = tf.boolean_mask(
+                point_clouds[frame_index],
+                point_clouds[frame_index, :, POINTCLOUD_LABEL_INDEX],
+                axis=0,
+            )
+            object_point_clouds = tf.boolean_mask(
+                additional_point_clouds[frame_index],
+                additional_point_clouds[frame_index, :, POINTCLOUD_LABEL_INDEX],
+                axis=0,
+            )
+
+            point_clouds_list += [
+                tf.concat([object_point_clouds, background_point_clouds], axis=0)
+            ]
+
+        point_clouds = tf.ragged.stack(point_clouds_list)
+        bounding_boxes = tf.RaggedTensor.from_tensor(
+            transformation[ADDITIONAL_BOUNDING_BOXES]
+        )
+        return (
+            point_clouds.to_tensor(shape=original_point_clouds_shape),
+            bounding_boxes.to_tensor(shape=original_bounding_boxes_shape),
+        )
+
+    def _augment(self, inputs):
+        result = inputs
+        point_clouds = inputs.get(POINT_CLOUDS, None)
```

Comment:
Is there a reason we can't just use bracket indexing here? e.g. inputs[POINT_CLOUDS]

Reply:
Updated